Source/WebKit/Platform/IPC/ArgumentCoders.h

```diff
@@ -75,33 +75,44 @@ template<typename T, size_t Extent> struct ArgumentCoder<Span<T, Extent>> {
     template<typename Encoder>
     static void encode(Encoder& encoder, const Span<T, Extent>& span)
     {
-        if constexpr (Extent == WTF::dynamic_extent)
-            encoder << static_cast<uint64_t>(span.size());
-        encoder.encodeFixedLengthData(reinterpret_cast<const uint8_t*>(span.data()), span.size() * sizeof(T), alignof(T));
+        if constexpr (Extent == WTF::dynamic_extent) {
+            auto size = static_cast<uint64_t>(span.size());
+            encoder << size;
+            if (!size)
+                return;
+        } else
+            static_assert(Extent, "Can't encode a fixed size of 0");
+
+        encoder.encodeSpan(span);
     }
+
     template<typename Decoder>
     static std::optional<Span<T, Extent>> decode(Decoder& decoder)
     {
-        std::optional<uint64_t> size;
+        std::optional<uint64_t> decodedSize;
```

Comment:
Regardless of whether you apply the behavior change above:
You could move the optional handling fully in the constexpr if.
So you have form 
```
size_t size = Extent;
if constexpr (...) {
  ...
  if (isInBounds...)
  ...
  size = *decodedSize;
} 
```

Reply:
True, will do.